monkey/monkey_island/cc/ui/src/components/Main.tsx

```diff
@@ -96,10 +96,23 @@ class AppComponent extends AuthComponent {
             if (this.state.islandMode === "unset") {
               return
             }
+
+            // check if any Agent was ever run
+            let any_agent_exists = false;
+            this.authFetch('/api/agents')
```

Comment:
The fact that we need to download the whole collection just to check if any agents ran is a bit disturbing. If continue this route we'll soon need redux/global store

Reply:
We discussed a bit over webex. This is ultimately a scalability concern and there are plans for addressing it.